File: Module_01_Big_O/Topic_03_Linear_Time_O(n)/Python/example_01.py
"""
Course title: AlgoMania
Instructor: Thiago Brito - Universo DEV
Example adapted by: Charles Fernandes de Souza
Date: September 27, 2024
"""
def print_items(my_array):
  for item in my_array:
        # Since it iterates through all elements once, the loop runs 'n' times,
        # where 'n' is the size of the array.
    print(item)

def main():
  print_items(my_array = [10,20,30,40])
main()

File: Module_01_Big_O/Topic_03_Linear_Time_O(n)/Java/Example01/src/com/souza/charles/Example01.java
package com.souza.charles;
/**
 Course title: AlgoMania
 Instructor: Thiago Brito - Universo DEV
 Example adapted by: Charles Fernandes de Souza
 Date: September 28, 2024
 */
public class Example01 {

    private static void printItem(int[] myArray) {
        for (int item : myArray){     // Since it iterates through all elements once, the loop runs 'n' times,
            System.out.println(item); // where 'n' is the size of the array.
        }
    }

    public static void main(String[] args) {
        int [] myArray = {10, 20, 30, 40};
        printItem(myArray);
    }
}
